nixos/modules/services/home-automation/zigbee2mqtt.nix

```
@@ -76,9 +76,7 @@ in
 
         # Hardening
         CapabilityBoundingSet = "";
-        DeviceAllow = [
-          config.services.zigbee2mqtt.settings.serial.port
-        ];
+        DeviceAllow = let port = cfg.settings.serial.port; in lib.optional (lib.hasPrefix "/dev/" port) port;
```

The `let` is a new evaluation context, which seems overkill here to me.

And next I'd like to use optionals where ever I can, since it uses brackets and is foolproof.

So I have a slight preference for going with

```suggestion
        DeviceAllow = lib.optionals (lib.hasPrefix "/dev/" cfg.settings.serial.port) [ cfg.settings.serial.port ];
```